### ISSUE TITLE
Accessibility issues on Design and Development page

### ISSUE THREAD

**Issue by chosak** [cfpb/design-system#1408]
**Describe the bug**

The D&D page fails Lighthouse accessibility checks, see full report [here](https://storage.googleapis.com/lighthouse-infrastructure.appspot.com/reports/1665447604309-92429.report.html):

1. Empty link (`<a></a>`) without any text generates "Links do not have a discernible name" error.
2. Some `<h5>` tags live directly under `<h3>` without an intervening `<h4>`, which causes "Heading elements are not in a sequentially-descending order".

@jenn-franklin this first one seems like some sort of weird side effect from Netlify CMS editing; the second one I'm less sure of how/if it can be fixed within the DS pages.

**To Reproduce**
1. Go to https://cfpb.github.io/design-system/guidelines/design-and-development
4. Open DevTools and run a Lighthouse report

**Comment by sonnakim** [comment]
@chosak looks like this was resolved, so I'll close this. Let me know if I should reopen.

<img width="1190" alt="image" src="https://github.com/cfpb/design-system/assets/9451445/0e80e349-1c24-447d-bee5-919a9e5405fa">
